src/Meshes/linesMesh.ts

```diff
@@ -39,7 +40,11 @@ export class LinesMesh extends Mesh {
      */
     public intersectionThreshold: number;
 
-    private _colorShader: ShaderMaterial;
+    private _lineMaterial: Material;
+
+    private _isShaderMaterial(shader: Material): shader is ShaderMaterial {

```

Comment:
```suggestion
    private _isShaderMaterial(shader: Material):  {

```

Reply:
I think that won't work because this method is used to let Typescript know that the parameter is a `ShaderMaterial` or not. If simply returning a `boolean`, I think we will have an error when using the parameter and assuming it is a `ShaderMaterial`.

For eg:
```typescript
        if (!this._isShaderMaterial(this._lineMaterial)) {
            return;
        }

        const define = "#define " + label;
        const index = this._lineMaterial.options.defines.indexOf(define);
```
`this._lineMaterial.options` won't work because Typescript does not know that `this._lineMaterial` is a `ShaderMaterial`. We would need to cast all usages as `as ShaderMaterial`: it's easier to have `_isShaderMaterial` returns a `shader is ShaderMaterial` type I think. I don't think there's a performance penality of doing it like that?

Reply:
@Popov72  is right. We used [guard function](https://www.typescriptlang.org/docs/handbook/advanced-types.html) for type check. If that method returns `boolean` casting is unavoidable forcing:
```
(this._lineMaterial as ShaderMaterial).options.defines.indexOf(define);
```
Surely we don't want that...